tasks/localizationTasks.ts

```diff
@@ -0,0 +1,174 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as gulp from 'gulp';
+import * as process from 'node:process';
+import * as minimist from 'minimist';
+import { spawnSync } from 'node:child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as util from 'node:util';
+import { EOL } from 'node:os';
+import { Octokit } from '@octokit/rest';
+
+type Options = {
+    userName: string;
+    email: string;
+    owner: string;
+    commitSha: string;
+    targetRemoteRepo: string;
+    baseBranch: string;
+};
+
+const localizationLanguages = ['cs', 'de', 'es', 'fr', 'it', 'ja', 'ko', 'pl', 'pt-br', 'ru', 'tr', 'zh-cn', 'zh-tw'];
+const locFiles = ['bundle.l10n.%s.json', 'package.nls.%s.json'];
+
+function getGeneratedLocalizationChanges(diffFilesAndDirectories: string[]): string[] {
+    if (diffFilesAndDirectories.length == 0) {
+        return [];
+    }
+
+    const allPossibleLocalizationFileNames = getAllPossibleLocalizationFileNames();
+    const changedLocFilesOrDirectory = [];
+
+    for (const diffFileOrDirectory of diffFilesAndDirectories) {
+        const stat = fs.statSync(diffFileOrDirectory);
+        if (
+            stat.isFile() &&
+            allPossibleLocalizationFileNames.some((locFileName) => path.basename(diffFileOrDirectory) === locFileName)
+        ) {
+            console.log(`${diffFileOrDirectory} is changed.`);
+            changedLocFilesOrDirectory.push(diffFileOrDirectory);
+        }
+
+        if (stat.isDirectory() && diffFileOrDirectory !== 'l10n') {
+            console.log('l10n is changed.');
+            changedLocFilesOrDirectory.push('l10n');
+        }
+    }
+
+    return changedLocFilesOrDirectory;
+}
+
+function getAllPossibleLocalizationFileNames(): string[] {
+    const files = [];
+    for (const lang of localizationLanguages) {
+        for (const file of locFiles) {
+            files.push(util.format(file, lang));
```

Comment:
should we verify that all the files exist here and throw if not?  I think it would be a problem if one of them doesn't exist right?

Reply:
Not sure if I understand your comment.
This one has just generated all the possible files here. e.g.
a combination of `['bundle.l10n.%s.json', 'package.nls.%s.json']` with `['cs', 'de', 'es', 'fr', 'it', 'ja', 'ko', 'pl', 'pt-br', 'ru', 'tr', 'zh-cn', 'zh-tw']`
Later the real file path comes from the result of `git diff --name-only` so the file should exist.